### PR TITLE
Android br se

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

Closes https://github.com/brave/brave-browser/issues/2019

### CODE REVIEW

File: components/brave_rewards/browser/BUILD.gn [hunk 1]
```diff
@@ -71,12 +76,24 @@ source_set("browser") {
...
+
     deps += [
       "//brave/vendor/bat-native-ledger",
       "//net",
       "//url",
-      # for one shot event
-      "//extensions/common",
     ]
+
+    if (!is_android) {
```
Comment (bridiver) [inline components/brave_rewards/browser/BUILD.gn]: can we move this above as an `else` https://github.com/brave/brave-core/pull/821/files#diff-0deb8db0aa3d14248a092d269a764b4dR45 to keep them together?

File: components/brave_rewards/browser/rewards_notification_service_impl.cc [hunk 2]
```diff
@@ -23,9 +23,9 @@
 namespace brave_rewards {
 
 RewardsNotificationServiceImpl::RewardsNotificationServiceImpl(Profile* profile)
-    : profile_(profile),
+    : profile_(profile)
 #if BUILDFLAG(ENABLE_EXTENSIONS)
-      extension_rewards_notification_service_observer_(
+      , extension_rewards_notification_service_observer_(
```
Comment (bridiver) [inline components/brave_rewards/browser/rewards_notification_service_impl.cc]: this is likely going to fail lint checks. Can we just move initialization into the constructor body?

File: components/resources/BUILD.gn [hunk 3]
```diff
@@ -1,37 +1,39 @@
 import("//tools/grit/grit_rule.gni")
 import("//brave/components/brave_rewards/browser/buildflags/buildflags.gni")
 
-grit("resources") {
-  source = "brave_components_resources.grd"
+if (!is_android) {
+  grit("resources") {
```
Comment (bridiver) [inline components/resources/BUILD.gn]: what is including this for you in android?

File: browser/ui/webui/brave_webui_source.cc [hunk 4]
```diff
@@ -6,8 +6,12 @@
 
 #include <map>
 
+#if !defined(OS_ANDROID)
 #include "brave/components/brave_rewards/resources/grit/brave_rewards_resources.h"
```
Comment (bridiver) [inline browser/ui/webui/brave_webui_source.cc]: I thought you were sharing the actual resources now. They are split up so it should be possible

File: browser/ui/webui/brave_webui_source.cc [hunk 5]
```diff
@@ -6,8 +6,12 @@
...
 #include "components/grit/brave_components_resources.h"
```
Comment (bridiver) [inline browser/ui/webui/brave_webui_source.cc]: these I know you won't be including now, but brave_rewards_resources.h should be good

File: browser/ui/webui/brave_rewards_ui.cc [hunk 6]
```diff
@@ -14,7 +14,12 @@
 #include "brave/components/brave_rewards/browser/rewards_notification_service_observer.h"
 #include "brave/components/brave_rewards/browser/rewards_service_factory.h"
 #include "brave/components/brave_rewards/browser/rewards_service_observer.h"
+#if !defined(OS_ANDROID)
 #include "brave/components/brave_rewards/resources/grit/brave_rewards_resources.h"
```
Comment (bridiver) [inline browser/ui/webui/brave_rewards_ui.cc]: same as below. You should be able to use brave_rewards_resources.h

File: components/brave_rewards/browser/BUILD.gn (hunk 1 above)
Comment (SergeyZhukovsky) [reply components/brave_rewards/browser/BUILD.gn]: but the above is not under `if (brave_rewards_enabled)`

**Pull request merged**